axelor-base/src/test/java/com/axelor/apps/base/test/TestCompanyDateService.java

```diff
@@ -0,0 +1,61 @@
+package com.axelor.apps.base.test;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.service.CompanyDateService;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestCompanyDateService {
+
+  private static final String DD_MM_YYYY_HH_MM = "dd/MM/yyyy HH:mm";
+  private static final String M_D_YYYY_H_MM_A = "M/d/yyyy, h:mm a";
+  private static final String DD_MM_YYYY = "dd/MM/yyyy";
+  private static final String M_D_YYYY = "M/d/yyyy";
+
+  protected CompanyDateService companyDateService;
+  protected Company usCompany;
+  protected Company frCompany;
+
+  @Before
+  public void prepare() {
+    companyDateService = new CompanyDateService();
+    usCompany = prepareCompany(Locale.US.toLanguageTag());
+    frCompany = prepareCompany(Locale.FRANCE.toLanguageTag());
+  }
+
+  @Test
+  public void testDateFormatter() throws AxelorException {
+    Assert.assertEquals(getDateTimeFormat(M_D_YYYY), getCompanyDateformat(usCompany));
+    Assert.assertEquals(getDateTimeFormat(DD_MM_YYYY), getCompanyDateformat(frCompany));
+  }
+
+  @Test
+  public void testDateTimeFormatter() throws AxelorException {
+    Assert.assertEquals(getDateTimeFormat(M_D_YYYY_H_MM_A), getCompanyDateTimeformat(usCompany));
+    Assert.assertEquals(getDateTimeFormat(DD_MM_YYYY_HH_MM), getCompanyDateTimeformat(frCompany));
+  }
+
+  protected Company prepareCompany(String languageTag) {
+    Company company = new Company();
+    company.setLocale(languageTag);
+    return company;
+  }
+
+  protected String getDateTimeFormat(String pattern) {
+    return DateTimeFormatter.ofPattern(pattern).format(LocalDateTime.now());
```

Comment:
I think it will not necessary cause issue, but to follow best practice we should avoid having a test that is different on every call. Please use a fixed LocalDate and LocalDateTime for the test.